api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -834,3 +834,50 @@ message RequestIDExtension {
   // Request ID extension specific configuration.
   google.protobuf.Any typed_config = 1;
 }
+
+// If this is used, the cluster will only operate on one of the possible upstream protocols (HTTP/1.1, HTTP/2).
+// If :ref:`http2_protocol_options <envoy_api_field_config.cluster.v3.Cluster.http2_protocol_options>` are
+// present, HTTP2 will be used, otherwise HTTP1.1 will be used.
+message ExplicitHttpConfig {
+  oneof protocol_config {
+    config.core.v3.Http1ProtocolOptions http_protocol_options = 1;
+
+    config.core.v3.Http2ProtocolOptions http2_protocol_options = 2;
+  }
+}
+
+// If this is used, the cluster can use either of the configured protocols, and
+// will use whichecer protocol was used by the downstream connection.
+message UseDownstreamHttpConfig {
```

Comment:
So one more question for the @envoyproxy/api-shepherds 

Do we want this in the HCM?
Right now we've got this thing pattern of upstream getting a per-network-filter config, which is why I've stuck this there, but realistically this creates hard deps from the Upstream config to the HCM config, which I'm not a big fan of.  This IMO is really the intersection of HTTP and upstream, not the intersection of HCM and upstream, but I can't think of a better place to put it so baring advice I guess I'll go with this?

Reply:
IMO it's fine to have this here, but I think if you want to put this in a new proto closer to the upstream configs that would be fine also?